api/v1alpha1/envoyproxy_metric_types.go

```diff
@@ -79,4 +79,10 @@ type ProxyOpenTelemetrySink struct {
 type ProxyPrometheusProvider struct {
 	// Disable the Prometheus endpoint.
 	Disable bool `json:"disable,omitempty"`
+	// Enable the compression on Prometheus endpoint.
+	EnableCompression bool `json:"enableCompression,omitempty"`
```

Comment:
I think we can make `CompressionLibrary` to `*string`?

Reply:
I am using compression type as suggested by @arkodg

Reply:
BTW, can you point out something like: `compression is not a silver bullet, it just a trade off. It will reduce the cost of network, but will increase the usage of cpu and memory.`

Reply:
sure @zirain

Reply:
this is done @zirain, please help check